File: Python/7py 2/inverse_slice.py
"""
|--------------------------------------------------------------------------
| Thinkful - List and Loop Drills: Inverse Slicer
|--------------------------------------------------------------------------
|
"""

def inverse_slice(items, a, b):
   return items[0:a] + items[b:]


def inverse_slice(items, a, b):
   del items[a:b]
   return items


def inverse_slice(items, a, b):
   return [n for n in items if n not in items[a:b]]


print(inverse_slice([12, 14, 63, 72, 55, 24], 2, 4), [12, 14, 55, 24])
print(inverse_slice([12, 14, 63, 72, 55, 24], 0, 3), [72, 55, 24])
print(inverse_slice(['Intuition', 'is', 'a', 'poor', 'guide', 'when', 'facing', 'probabilistic', 'evidence'], 5, 13), 
      ['Intuition', 'is', 'a', 'poor', 'guide'])

File: PHP/7php 2/inverse_slice.php
<?php
/*
|--------------------------------------------------------------------------
| Thinkful - List and Loop Drills: Inverse Slicer
|--------------------------------------------------------------------------
|
*/

function inverse_slice($items, $a, $b) {
   return array_merge(array_slice($items, 0, $a), array_slice($items, $b));
}


function inverse_slice(array $items, int $a, int $b): array {
   $res = [];
   for ($i = 0; $i < sizeof($items); $i++) {
      if ($i < $a || $i >= $b) $res[] = $items[$i];
   }
   return $res;
}


function inverse_slice(array $items, int $a, int $b): array {
   array_splice($items,$a,$b-$a);
   return $items;
}


print_r(inverse_slice([12, 14, 63, 72, 55, 24], 2, 4));
print_r(inverse_slice([12, 14, 63, 72, 55, 24], 0, 3));
print_r(inverse_slice(["Intuition", "is", "a", "poor", "guide", "when", "facing", "probabilistic", "evidence"], 5, 13));